### PR TITLE
Preserve chain data

### PR DESCRIPTION
**Pull request by liayoo**
- Updated default chain data directory to ~/.ain/chains
- Introduced DATA_DIR env var (default = ~/.ain)
- Enabled data sharing between nodes by preserving chain data (reduces syncing time!)

### CODE REVIEW

File: deploy_blockchain_gcp.sh [hunk 1]
```diff
@@ -39,9 +39,6 @@ fi
 FILES_FOR_TRACKER="blockchain/ client/ common/ consensus/ db/ genesis-configs/ logger/ tracker-server/ package.json setup_tracker_gcp.sh setup_blockchain_ubuntu.sh start_tracker_gcp.sh"
 FILES_FOR_NODE="blockchain/ client/ common/ consensus/ db/ json_rpc/ genesis-configs/ logger/ node/ tx-pool/ p2p/ package.json setup_node_gcp.sh setup_blockchain_ubuntu.sh start_node_gcp.sh"
 
-printf "\nRemoving redundant files..."
-rm -rf chains logs
```
Comment (platfowner) [inline deploy_blockchain_gcp.sh:43]: What about applying to only incremental script for now?

Comment (platfowner) [reply deploy_blockchain_gcp.sh:43]: Ah, this is about local files. Please ignore my previous comment.

File: common/constants.js [hunk 2]
```diff
@@ -63,7 +64,11 @@ if (!semver.valid(CONSENSUS_PROTOCOL_VERSION)) {
   throw Error('Wrong data version format is specified for CONSENSUS_PROTOCOL_VERSION');
 }
 const LOGS_DIR = path.resolve(__dirname, '../logs');
-const CHAINS_DIR = path.resolve(__dirname, '../chains');
+const DATA_DIR = process.env.DATA_DIR || path.join(os.homedir(), '.ain');
```
Comment (platfowner) [inline common/constants.js:67]: What about not using dot(.)? It's hiding dirs and files in linux system.

What about using longer names?
- DATA_DIR -> BLOCHAIN_DATA_DIR
- '.ain' -> 'blockchain_data' or 'ain_blockchain_data'

File: deploy_blockchain_gcp.sh (hunk 1 above)
Comment (liayoo) [reply deploy_blockchain_gcp.sh:43]: Yes, it seemed we no longer scp the whole chains / logs, so this seemed unnecessary!

File: common/constants.js (hunk 2 above)
Comment (liayoo) [reply common/constants.js]: Updated the names & removed the dot!

**Review by liayoo — COMMENTED** [review]
Addressed all comments, PTAL @platfowner

File: common/constants.js [hunk 3]
```diff
@@ -63,7 +64,11 @@ if (!semver.valid(CONSENSUS_PROTOCOL_VERSION)) {
   throw Error('Wrong data version format is specified for CONSENSUS_PROTOCOL_VERSION');
 }
 const LOGS_DIR = path.resolve(__dirname, '../logs');
-const CHAINS_DIR = path.resolve(__dirname, '../chains');
+const BLOCHAIN_DATA_DIR = process.env.BLOCHAIN_DATA_DIR || path.join(os.homedir(), 'ain_blockchain_data');
```
Comment (platfowner) [inline common/constants.js:67]: minor: Using the project directory for blockchain data seems to be often natural and intuitive. So, could you consider alternative options below? 
- Use `../ain_blockchain_data` (or just `../blockchain_data`) as the default value
- Add setting for `BLOCHAIN_DATA_DIR` in scripts such as `start_servers.sh` and `start_servers_afan.sh` instead if necessary?

File: stop_servers.sh [hunk 4]
```diff
@@ -1,5 +1,4 @@
-
+killall -9 node # SIGKILL
+rm -rf ~/ain_blockchain_data/
```
Comment (platfowner) [inline stop_servers.sh:2]: Could you update `start_servers_afan.sh` and `reset_blockchain_gcp.sh` as well (or just remove the `rm -rf` line)?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments left.
Thanks!

File: common/constants.js (hunk 3 above)
Comment (liayoo) [reply common/constants.js]: Updated the default & added BLOCKCHAIN_DATA_DIR in start servers scripts!

File: stop_servers.sh (hunk 4 above)
Comment (liayoo) [reply stop_servers.sh:2]: Done!

**Comment by liayoo** [comment]
Thanks for the review & suggestions! Now merging..

File: start_servers.sh [hunk 5]
```diff
@@ -1,33 +1,33 @@
 # PARENT CHAIN
 node ./tracker-server/index.js &
 sleep 5
-MIN_NUM_VALIDATORS=5 ACCOUNT_INDEX=0 STAKE=100000 CONSOLE_LOG=true ENABLE_DEV_SET_CLIENT_API=true ENABLE_TX_SIG_VERIF_WORKAROUND=true ENABLE_GAS_FEE_WORKAROUND=true node ./client/index.js &
+MIN_NUM_VALIDATORS=5 ACCOUNT_INDEX=0 STAKE=100000 CONSOLE_LOG=true ENABLE_DEV_SET_CLIENT_API=true ENABLE_TX_SIG_VERIF_WORKAROUND=true ENABLE_GAS_FEE_WORKAROUND=true BLOCKCHAIN_DATA_DIR=~/ain_blockchain_data node ./client/index.js &
 sleep 10
-MIN_NUM_VALIDATORS=5 ACCOUNT_INDEX=1 STAKE=100000 CONSOLE_LOG=true ENABLE_DEV_SET_CLIENT_API=true ENABLE_TX_SIG_VERIF_WORKAROUND=true ENABLE_GAS_FEE_WORKAROUND=true node ./client/index.js &
+MIN_NUM_VALIDATORS=5 ACCOUNT_INDEX=1 STAKE=100000 CONSOLE_LOG=true ENABLE_DEV_SET_CLIENT_API=true ENABLE_TX_SIG_VERIF_WORKAROUND=true ENABLE_GAS_FEE_WORKAROUND=true BLOCKCHAIN_DATA_DIR=~/ain_blockchain_data node ./client/index.js &
+sleep 1
+MIN_NUM_VALIDATORS=5ACCOUNT_INDEX=2 STAKE=100000 CONSOLE_LOG=true ENABLE_DEV_SET_CLIENT_API=true ENABLE_TX_SIG_VERIF_WORKAROUND=true ENABLE_GAS_FEE_WORKAROUND=true BLOCKCHAIN_DATA_DIR=~/ain_blockchain_data node ./client/index.js &
```
Comment (platfowner) [inline start_servers.sh:8]: space around '5'?

**Pull request merged**